File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/AutorBO.java

package pe.edu.pucp.softbiblioteca.rmi.interfaces;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;
import pe.edu.pucp.softbiblioteca.publicaciones.model.Autor;

public interface AutorBO extends Remote{
    
    public ArrayList<Autor> buscarAutorPorNombre(String nombre, Integer limite) throws RemoteException;
    
    public Integer insertarAutor(String nombre) throws RemoteException;
    
    public ArrayList<Autor> listarTodos_AutoresDeLibroPorID(Integer idLibro) throws RemoteException;
    
    public ArrayList<Autor> listarTodos_AutoresDeLibro(Integer idLibro, String titulo, String fechaPublicacion,
                                                       String descripcion, Integer hayFormatoFisico, Integer hayFormatoVirtual,
                                                       Integer idBibliotecaAsociada, Integer idEditorial, String tipo, Integer volumen,
                                                       String materia, String genero, Integer tomo, String nombreCompleto) throws RemoteException;
    
    public ArrayList<Autor> listarTodos_AutoresDeTesisPorID(Integer idTesis) throws RemoteException;
    
    public ArrayList<Autor> listarTodos_AutoresDeTesis(Integer idTesis,String titulo, String fechaPublicacion,
                                                       String descripcion, Integer hayFormatoFisico, Integer hayFormatoVirtual,
                                                       Integer idBibliotecaAsociada, String gradoAcademico, String nombreCompleto) throws RemoteException;
    
    public Integer insertar(String nombreCompleto) throws RemoteException;
    
    public Integer modificar(Integer idAutor,String nombreCompleto) throws RemoteException;
    
    public Integer eliminar(Integer idAutor) throws RemoteException;
    
    public Autor obtenerPorId(Integer idAutor) throws RemoteException;
    
}

File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaPublicacionBO/AutorBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;

namespace SoftBibliotecaPublicacionBO
{
    public class AutorBO : BaseBO
    {
        /*
        public int insertar(string nombreCompleto)
        {
            return this.WsCliente.autor_insertar(nombreCompleto);
        }

        public int modificar(int idAutor, string nombreCompleto)
        {
            return this.WsCliente.autor_modificar(idAutor,nombreCompleto);
        }

        public int eliminar(int idAutor)
        {
            return this.WsCliente.autor_eliminar(idAutor);
        }

        public BindingList<autor> listarTodos()
        {
            try
            {
                autor[] autores = this.WsCliente.autor_listarTodos();
                if(autores == null) return new BindingList<autor>();
                return new BindingList<autor>(autores);
            }
            catch (Exception e)
            {
                return new BindingList<autor>();
            }
        }

        public BindingList<autor> listarTodos(string nombreCompleto)
        {
            try
            {
                autor[] autores = this.WsCliente.autor_listarTodos_porParametros(nombreCompleto);
                if (autores == null) return new BindingList<autor>();
                return new BindingList<autor>(autores);
            }
            catch (Exception e)
            {
                return new BindingList<autor>();
            }
        }

        public autor obtenerPorId(int idAutor)
        {
            return this.WsCliente.autor_obtenerPorId(idAutor);
        }

        */
        public BindingList<autor> buscarAutorPorNombre(string nombre, int limite)
        {

            try
            {
                autor[] autores = this.WsCliente.buscarAutorPorNombre(nombre,limite);
                if (autores == null) return new BindingList<autor>();
                return new BindingList<autor>(autores);

            }
            catch (Exception ex)
            {
                return new BindingList<autor>();
            }
        }
        public int insertarAutor(String nombre)
        {
            return this.WsCliente.insertarAutor(nombre);
        }
        public BindingList<autor> listarTodos_AutoresDeLibroPorID(int id)
        {
            try
            {
                autor[] autores = this.WsCliente.listarTodos_AutoresDeLibroPorID(id);
                if (autores == null) return new BindingList<autor>();
                return new BindingList<autor>(autores);

            }
            catch (Exception ex)
            {
                return new BindingList<autor>();
            }
        }
        public BindingList<autor> listarTodos_AutoresDeLibro(int idLibro, string titulo, string fechaPublicacion,
                                                       string descripcion, int hayFormatoFisico, int hayFormatoVirtual,
                                                       int idBibliotecaAsociada, int idEditorial, string tipo, int volumen,
                                                       string materia, string genero, int tomo, string nombreCompleto)
        {
            try
            {
                autor[] autores = this.WsCliente.listarTodos_AutoresDeLibro(idLibro, titulo, fechaPublicacion, descripcion, hayFormatoFisico, hayFormatoVirtual, idBibliotecaAsociada, idEditorial, tipo, volumen, materia, genero, tomo, nombreCompleto);
                if (autores == null) return new BindingList<autor>();
                return new BindingList<autor>(autores);

            }
            catch (Exception ex)
            {
                return new BindingList<autor>();
            }

        }
        public BindingList<autor> listarTodos_AutoresDeTesisPorID(int id)
        {
            try
            {
                autor[] autores = this.WsCliente.listarTodos_AutoresDeTesisPorID(id);
                if (autores == null) return new BindingList<autor>();
                return new BindingList<autor>(autores);

            }
            catch (Exception ex)
            {
                return new BindingList<autor>();
            }
        }
        public BindingList<autor> listarTodos_AutoresDeTesis(int idTesis, string titulo, string fechaPublicacion,
                                                      string descripcion, int hayFormatoFisico, int hayFormatoVirtual,
                                                      int idBibliotecaAsociada, string gradoAcademico, string nombreCompleto)
        {
            try
            {
                autor[] autores = this.WsCliente.listarTodos_AutoresDeTesis(idTesis, titulo, fechaPublicacion, descripcion, hayFormatoFisico, hayFormatoVirtual, idBibliotecaAsociada, gradoAcademico, nombreCompleto);
                if (autores == null) return new BindingList<autor>();
                return new BindingList<autor>(autores);

            }
            catch (Exception ex)
            {
                return new BindingList<autor>();
            }

        }
    }
}
